lib/executeHook.js

```diff
@@ -16,51 +16,68 @@ const formatError = function(error) {
 
 // returns an async op that will execute the hooks
 module.exports = function(server, settings, collection, hook) {
-  return {
-    // log that it's underway:
-    checkRetryCount(done) {
-      if (hook.runCount && hook.runCount >= settings.maxRetries) {
-        collection.update({ _id: hook._id }, { $set: { status: 'aborted' } }, (err) => {
+  const launchHook = (hookLaunchedCallback) => {
```

Comment:
This shouldn't have to change at all

Reply:
Query hooks should do the logging and pending check. If auto map causes issues, you can just switch with auto inject